HireUp-Turing/HireUp_frontend
# Feature/ge inbox

#### Type of Change Made 
- [ ] formatting 

#### What's this PR do?
This PR wraps up the functionality for creating a new applicant by routing to the new applicant form, posting a new applicant, and redirecting that new applicant based on applicant id and conditionally renders the header links if a user is 'authorized', which mimics being logged in by providing a check to see if there is an id present or undefined. 

Creates an Inbox components that is accessible to an auth/logged in applicant via the inbox link. Fetched messages are displayed and if clicked, styling indicates the message has been read. 

A user generator is created to fill our page with unique and fun applicants! Nice work, Greyson!

#### Where should the reviewer start?
Heart of changes below, but many files were touched with this PR.
 src/assets/api-calls.tsx
src/assets/user-maker.tsx
src/contexts/index.tsx
src/components/Inbox/Inbox.tsx

#### How should this be manually tested?
Go to applicant form and create and applicant and see new applicant displayed.
Go to http://localhost:3000/applicant/1/inbox to view messages for applicant based on id.
Click on a message and watch it change!

#### Any background context you want to provide?
Posting if a message is read will be a future extension. 

#### What are the relevant tickets?
closes #43 
closes #45 
closes #47
